pkg/networkpolicy/packet.go

```diff
@@ -55,33 +57,51 @@ func parsePacket(b []byte) (packet, error) {
 		copy(t.srcIP, b[8:24])
 		t.dstIP = make(net.IP, net.IPv6len)
 		copy(t.dstIP, b[24:40])
-		// NextHeader (not extension headers supported)
-		protocol = int(b[6])
+		// Handle extension headers. TODO: The recommended order is assumed!
+		nxtHeader = int(b[6])
+		l4offset = hdrlen
+		for nxtHeader == syscall.IPPROTO_DSTOPTS || nxtHeader == syscall.IPPROTO_HOPOPTS || nxtHeader == syscall.IPPROTO_ROUTING {
+			// These header have a lenght in 8-octet units, not including the first 8 octets
+			nxtHeader = int(b[l4offset])
```

Comment:
we are capping the packet , should we need to validate `len(b)` to avoid a panic?

Reply:
Yes, it should be checked in every step. I plan to make a unit-test that feeds capped packets from 1 byte to 128, and verify that no segv occurs.